docs/docs/migrations.md

```diff
@@ -306,6 +306,74 @@ queryInterface.removeIndex('Person', 'SuperDuperIndex')
 queryInterface.removeIndex('Person', ['firstname', 'lastname'])
 ```
 
+### addConstraint(tableName, indexNameOrAttributes, options)
+This method adds a new constraint of the specified type.
+
+Available constraints:
+ - UNIQUE
+ - DEFAULT (MSSQL only)
+ - CHECK (MySQL - Ignored by the database engine )
+ - FOREIGN KEY
+ - PRIMARY KEY
+
+```js
+//UNIQUE
+queryInterface.addConstraint('Users', ['email'], {
+  type: 'unique',
+  name: 'custom_unique_constraint_name' //If not specifed, sequelize automatically creates a named constraint based on constraint type, table & column names
+});
+
+//CHECK
+queryInterface.addConstraint('Users', ['roles'], {
+  type: 'check',
+  where: {
+    roles: ['user', 'admin', 'moderator', 'guest']
+  }
+});
+//Constraint name is automatically constructed.
+
+//Default - MSSQL only
+queryInterface.addConstraint('Users', ['roles'], {
+  type: 'default',
+  defaultValue: 'guest'
+});
+
+//Primary Key
+queryInterface.addConstraint('Users', ['username'], {
+  type: 'primary key',
+  name: 'custom_primary_constraint_name'
+});
+
+//Foreign Key
+queryInterface.addConstraint('Posts', ['username'], {
+  type: 'FOREIGN KEY',
+  references: { //Required field
+    model: 'target_table_name',
```

Comment:
@sushantdhiman Since constraints are executed only via migrations, i'll keep it to tableName only for now. Even `createTable` method expects a string and not a [model](https://github.com/sequelize/sequelize/blob/master/lib/dialects/postgres/query-generator.js#L444-L467)